plan.go

```diff
@@ -247,40 +252,28 @@ func extractSelectionSet(ctx *PlanningContext, insertionPoint []string, parentTy
 					continue
 				}
 				implementationType := ctx.Schema.Types[implementationName]
-				possibleId := &ast.InlineFragment{
-					TypeCondition: implementationName,
-					SelectionSet: []ast.Selection{
-						&ast.Field{
-							Alias:      "_id",
-							Name:       "id",
-							Definition: implementationType.Fields.ForName("id"),
-						},
-					},
-					ObjectDefinition: implementationType,
+
+				if idDef := implementationType.Fields.ForName("id"); idDef != nil {
+					possibleId := &ast.InlineFragment{
+						TypeCondition: implementationName,
+						SelectionSet: []ast.Selection{&ast.Field{Alias: "_bramble_id", Name: "id", Definition: idDef}},
+						ObjectDefinition: implementationType,
+					}
+					selectionSetResult = append(selectionSetResult, possibleId)
 				}
-				selectionSetResult = append([]ast.Selection{possibleId}, selectionSetResult...)
 				break
 			}
 		}
-	// Otherwise, add an id selection to boundary types where the result
-	// will be merged with another step (i.e.: has children or is a child step).
-	} else if parentType != queryObjectName &&
-		parentType != mutationObjectName &&
-		ctx.IsBoundary[parentType] &&
-		(childstep || len(childrenStepsResult) > 0) &&
-		parentDef.Fields.ForName("id") != nil &&
-		!selectionSetHasFieldNamed(selectionSetResult, "id") {
-		id := &ast.Field{
-			Alias:      "_id",
-			Name:       "id",
-			Definition: parentDef.Fields.ForName("id"),
```

Comment:
This check has been greatly simplified – any boundary object with an `id` field defined gets a `_bramble_id` hint added to the end of the selection. It's more verbose, but it's extremely predictable and consistent.